gateway/gateway.go

```diff
@@ -318,16 +388,13 @@ func (gateway *HandleT) userWebRequestBatchDBWriter(dbWriterWorker *dbWriterWork
 			jobWriteKeyMap[newJob.UUID] = writeKey
 			jobEventCountMap[newJob.UUID] = totalEventsInReq
 		}
-		var errorMessagesMap map[uuid.UUID]string
-		gwAllowPartialWriteWithErrors := config.GetBool("Gateway.allowPartialWriteWithErrors", true)
-		switch gwAllowPartialWriteWithErrors {
-		case true:
-			errorMessagesMap = gateway.jobsDB.StoreWithRetryEach(jobList)
-		case false:
-			gateway.jobsDB.Store(jobList)
+
+		gateway.userWorkerBatchRequestQ <- &userWorkerBatchRequestT{jobList: jobList,
+			allMessageIdsSet: allMessageIdsSet,
+			respChannel:      userWebRequestWorker.reponseQ,
 		}
 
-		gateway.writeToBadger(allMessageIdsSet)
+		errorMessagesMap := <-userWebRequestWorker.reponseQ
```

Comment:
We are waiting for each batch to be written before giving the next batch is given to dbWriterWorkerProcess. Instead, could we not give all the batches available to us and let dbWriteWorkerProcess do its own batching before writing?

Reply:
Will work on improvising when need arises.